pkg/deduperkey/key.go

```diff
@@ -41,6 +44,25 @@ type Key struct {
 	ResourceType reflect.Type
 }
 
+// ParseKeySlice returns a list of Key objects from a list a string formatted keys. An error returned means that some
+// of the keys might have failed when being parsed.
+func ParseKeySlice(keys []string) ([]Key, error) {
+	if keys == nil {
```

Comment:
This check isn't necessary so I think it's better to omit it

Reply:
Done